src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -1061,6 +1062,25 @@ public void onIndexModule(IndexModule module) {
         });
     }
 
+    /**
+     * Finds the suggested number of indexing threads based on the number of available processors
+     *
+     * @return suggested number of indexing threads
+     */
+    public static int getHardwareDefaultIndexThreadQty() {
```

Comment:
make it package-private as we aren't using it outside of this class except in tests